code/app/src/main/java/com/adobe/marketing/edge/identity/app/ui/CustomIdentityFragment.kt

```diff
@@ -93,6 +104,44 @@ class CustomIdentityFragment : Fragment() {
             Identity.removeIdentity(item, namespace)
         }
 
+        // Button for Set Ad ID behavior
+        // Sets the advertising identifier set in the corresponding textfield using the MobileCore API
+        root.findViewById<Button>(R.id.btn_set_ad_id).setOnClickListener {
+            val adId: String? = sharedViewModel.adId.value
+            MobileCore.setAdvertisingIdentifier(adId)
+        }
+
+        // For details on implementation tips and differences between Google Play Services Ads vs AndroidX Ads,
+        // please see the project Documentation -> AEPEdgeIdentity.md : Test App -> GAID Implementation
+        root.findViewById<Button>(R.id.btn_get_gaid).setOnClickListener {
+            val context = context
+            Log.d("Custom_Identity_Fragment", "context: $context, appContext: ${context?.applicationContext}")
+            if (context != null) {
+                // Implementation for google play services ads
+                // Logic implemented in ViewModel to leverage managed Kotlin coroutine scope from lifecycle-viewmodel-ktx
+                // lifecycle-viewmodel-ktx requires compiled sdk version 31+
+                GlobalScope.launch {
+                    sharedViewModel.getGAID(context.applicationContext)
+                }
+            }
```

Comment:
Is this button intended to fetch the device advertising id or the adid currently set with the SDK?

Reply:
Implementation has been updated so that the getter function is a pure getter, and the returned ad ID is sent through the MobileCore.setAdvertisingIdentifier